topic-operator/src/main/java/io/strimzi/operator/topic/Config.java

```diff
@@ -107,6 +107,14 @@ private Value(String key, Type<? extends T> type, boolean required) {
     public static final String TC_TLS_KEYSTORE_LOCATION = "STRIMZI_KEYSTORE_LOCATION";
     public static final String TC_TLS_KEYSTORE_PASSWORD = "STRIMZI_KEYSTORE_PASSWORD";
 
+    /* config for SASL_SSL authentication */
+    public static final String TC_TLS_SASL_ENABLED = "STRIMZI_TLS_SASL_ENABLED";
```

Comment:
I am unsure whether this was discussed/agreed on in another discussion, but are we sure `STRIMZI_TLS_SASL_ENABLED` as a boolean is required, can we not infer it being enabled if `TC_TLS_SASL_MECHANISM` is a non empty `String`?

Reply:
Yeah, this was discussed during my work on the user operator; I agree we don't need an extra variable.
Even in the topic operator we have the `TLS_ENABLED` which could be removed as well, because it could be inferred by the corresponding `TLS_TRUSTSTORE....` and `TLS_KEYSTORE...` but it's not something for the current PR.